src/ExtendedPdo.php

```diff
@@ -83,17 +99,24 @@ public function __construct(
 
         // set quotes for identifier names
         $this->setQuoteName($parts[0]);
+
+        // create a connection immediately
+        if (isset($options[static::CONNECT_IMMEDIATELY]) && $options[static::CONNECT_IMMEDIATELY]) {
+            $this->establishConnection();
+        }
     }
 
     public static function connect(
         string $dsn,
         ?string $username = null,
         ?string $password = null,
-        ?array $options = [],
+        ?array $options = null,
```

Comment:
can't we actually use [] instead of null, this will also remove the usage of `$options ??= [];` right ?